### PR TITLE
Feat/posts

### PR DESCRIPTION
**Pull request by bhavyaj12**
### Added new post, fetch posts and delete post functionality

- Added add post, fetch post, delete post functionality using redux - createSlice and asyncThunk
- Added post with image input
- Added postsSlice with reducers
- Added Fetch posts functionality in user home (pages)
- Added delete post functionality for user's own post

**Please check app is RESPONSIVE**

### What not to check
- Edit functionality

### CODE REVIEW

**Comment by junaidshaikh-js** [comment]
suggestions
- Show recent posts at the top
- Give text area fix height and scroll the overflow

File: src/components/AddPost.jsx [hunk 1]
```diff
@@ -1,12 +1,65 @@
...
+    const postData = {
+      firstName,
+      lastName,
+      content: addContent,
+      postImage: addPostImage,
+    };
+
+    try {
+      const response = await dispatch(addPost({ token, postData }));
+      console.log("response from add post handler", response);
```
Comment (junaidshaikh-js) [inline src/components/AddPost.jsx:29]: remove console logs

**Review by junaidshaikh-js — APPROVED** [review]
The code looks clean and well structured. Can be merged.

**Review by vikrantkode — APPROVED** [review]
The code looks clean 
well structured
Functionality is working fine as mentioned.
Can be merged.

**Pull request merged**